contactsummary/CRM/Contactsummary/Utils/Absences.php

```diff
@@ -1,69 +1,61 @@
-  const TYPE_VACATION = 'vacation';
 
   /**
-   * todo: review if this method is needed at all
-   *
-   * @param array $absenceTypes
-   * @param       $periodId
-   *
-   * @return int
-   */
-  public static function getTotalAbsences($absenceTypes = array(), $periodId) {
-    $total = static::getAbsenceDuration($absenceTypes, $periodId);
-
-    return $total;
-  }
-
-  /**
-   * Get duration of absences, in minutes.
-   *
-   * @param array $absenceTypeNames
-   * @param       $periodId
+   * Get duration of absences for specific absence type, in minutes.
    *
+   * @param $absenceTypeName
+   * @param $periodId
    * @return int
    */
-  private static function getAbsenceDuration($absenceTypeNames = array(), $periodId) {
-    $absenceTypeIds = static::getActivityIdsForAbsenceTypeNames($absenceTypeNames, $periodId);
-    $activityStatuses = CRM_HRAbsence_BAO_HRAbsenceType::getActivityStatus('name');
+  public static function getTotalAbsences($absenceTypeName, $periodId) {
 
     $sql = "
-      SELECT SUM(a1.duration) duration
+      SELECT SUM(duration) AS duration from (
+      SELECT a.duration
       FROM civicrm_activity a
-
-      INNER JOIN civicrm_activity_contact ac
-      ON a.id = ac.activity_id
-
-      INNER JOIN civicrm_contact c
-      ON (ac.contact_id = c.id AND c.contact_type = 'Individual')
-
-      INNER JOIN civicrm_hrjobcontract jc
-      ON c.id = jc.contact_id AND jc.is_primary = 1
-
-      INNER JOIN civicrm_hrjobcontract_revision jcr
-      ON (jc.id = jcr.jobcontract_id AND jcr.effective_date <= NOW())
-
-      INNER JOIN civicrm_hrjobcontract_details jcd
-      ON (jcr.id = jcd.jobcontract_revision_id AND (jcd.period_end_date >= NOW() OR jcd.period_end_date IS NULL))
-
-      INNER JOIN civicrm_activity a1
-      ON (a.id = a1.source_record_id AND a1.activity_type_id = %1 AND a1.status_id = %2)
+      LEFT JOIN civicrm_activity a2 ON a.source_record_id = a2.id
+      LEFT JOIN civicrm_activity_contact ac ON a2.id = ac.activity_id
+      LEFT JOIN civicrm_contact c ON ac.contact_id = c.id
+      LEFT JOIN civicrm_hrjobcontract hrjc ON (c.id = hrjc.contact_id)
+      LEFT JOIN civicrm_hrjobcontract_revision hrjr ON hrjr.jobcontract_id = hrjc.id
+      LEFT JOIN civicrm_hrjobcontract_details hrjd ON hrjr.details_revision_id = hrjd.jobcontract_revision_id
 
       WHERE
-        a.id IN (" . implode(',', $absenceTypeIds) . ")
-        AND a.status_id = %2
+      a.source_record_id
+      IN (
+      SELECT t2.id
+      FROM civicrm_activity_contact t1
+      LEFT JOIN civicrm_activity t2 ON t1.activity_id = t2.id
+      WHERE t2.activity_type_id = %1
+      AND t1.record_type_id = 3
+      )
+
+      AND a.status_id = %2
+      AND (a.activity_date_time >= %3 AND a.activity_date_time < %4)
+      AND a.is_deleted = 0
+
+      AND ac.record_type_id =3
+
+      AND c.contact_type = 'Individual'
+
+      AND hrjc.deleted = 0
+      AND hrjr.deleted = 0
+      AND hrjd.period_start_date <= CURDATE()
```

Comment:
Is it possible to use `CURDATE()` only once ? maybe pass it as a param. Not sure how much performace effective this is but just a thought :)